lib/puppet/configurer/fact_handler.rb

```diff
@@ -32,6 +32,6 @@ def facts_for_uploading
 
     text = facts.render(:pson)
 
-    {:facts_format => :pson, :facts => CGI.escape(text)}
+    {:facts_format => :pson, :facts => Puppet::Util.uri_query_escape(text)}
```

Comment:
Bummer... I was hoping to get this one in first ;0

`CGI.escape` is *mostly* correct, except when it comes to spaces - hence our own helper ;(